projects/igniteui-angular/src/lib/navigation-drawer/navigation-drawer.component.ts

```diff
@@ -59,61 +59,184 @@ export class IgxNavigationDrawerComponent implements
 
     @HostBinding('class') public cssClass = 'igx-nav-drawer';
 
-    /** ID of the component */
+    /**
+     * ID of the component
+     *
+     * ```typescript
+     * // get
+     * let myNavDrawerId = this.navdrawer.id;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     *  <igx-nav-drawer id='navdrawer'></igx-nav-drawer>
+     * ```
+     */
     @HostBinding('attr.id')
     @Input() public id = `igx-nav-drawer-${NEXT_ID++}`;
 
     /**
-     * Position of the Navigation Drawer. Can be "left"(default) or "right". Only has effect when not pinned.
+     * Position of the Navigation Drawer. Can be "left"(default) or "right".
+     * Only has effect when not pinned.
+     *
+     * ```typescript
+     * // get
+     * let myNavDrawerPosition = this.navdrawer.position;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-nav-drawer [position]='left'></igx-nav-drawer>
```

Comment:
For this to be passed as string either `position='left'` or `[position]="'left'"`.
Square brackets will evaluate the value as JS/template variable.